Cython/Compiler/Options.py

```diff
@@ -376,6 +376,20 @@ def normalise_encoding_name(option_name, encoding):
 }
 
 
+# a list of directives that (when used as a decorator) are only applied to
+# the object they decorate and not to its children.
+immediate_decorator_directives = (
+    'cfunc', 'ccall', 'cclass',
+    # function signature directives
+    'inline', 'exceptval', 'returns',
+    # class directives
+    'freelist', 'no_gc', 'no_gc_clear', 'type_version_tag', 'final',
+    'auto_pickle', 'internal',
+    # testing directives
+    'test_fail_if_path_exists', 'test_assert_path_exists',
```

Comment:
Having these here possibly removes the need for `copy_inherited_directives`. I don't know if it's worth going back and removing that to see if it genuinely isn't needed any more

Reply:
I've tried to remove it. I'll revert the change if it doesn't work

Reply:
Couldn't easily get it to work so I'll leave it for now and revert back to the version that does work.